src/assets/styles/components/playlist.scss

```diff
@@ -0,0 +1,213 @@
+@import 'compass/css3/flexbox';
+
+@mixin listLayout() {
+    .cld-plw-layout {
+        &.cld-plw-vertical {
+            @include flex-direction(column);
+            
+            .cld-plw-col-list {
+                width: inherit !important;
+            }
+
+            .cld-plw-panel-item {
+                background: none !important;
+
+                img {
+                    display: block;
+                    width: 150px;
+                }
+
+                .cld-plw-item-info-wrap {
+                    left: 160px;
+                }
+
+                .cld-plw-item-title-next {
+                    display: block;
+                }
+            }
+        }
+    }
+}
+
+.cld-plw-horizontal {
+    @include display-flex(inline-flex);
+    @include flex-direction(row);
+
+    .cld-plw-col-list {
+        @include display-flex(flex);
+        @include flex-direction(row);
+        padding-top: 4px;
+    } 
+
+    .cld-plw-panel {
+        overflow: hidden;
+        @include flex(1);
+        @include display-flex(flex);
+        @include flex-direction(row);
+    
+    }
+
+    .cld-plw-panel-item {
+        border-width: 1px;
+        border-style: solid;
+        transition: border-color .25s;
+        margin: 2px;
+        background-image: none;
+
+        &:first-child {
+            margin-left: 0;
+        }
+
+        &:last-child {
+            margin-right: 0;
+        }
+    }   
+}
+
+
+.cld-plw-vertical {
+
+    @include display-flex(inline-flex);
+    @include flex-direction(row);
+
+    &.cld-plw-layout {
+        > div:first-child {
+            flex: 1;
+        }
+    }
+
+    .cld-plw-col-list {
+        @include display-flex(flex);
+        @include flex-direction(column);
+    } 
+
+    .cld-plw-col-player {
+        padding-right: 4px;
+    }
+
+    .cld-plw-panel {
+        @include flex(1);
+        @include display-flex(flex);
+        @include flex-direction(column);
+    }
+
+    .cld-plw-panel-item {
+        
+        margin-bottom: 4px;
+        
+        &:last-child {
+            margin-bottom: 0;
+        }
+
+    }
+}
+
+.cld-flex-col-1 {
+    flex: 0 1 100%;
+}
+
+.cld-flex-col-2 {
+    flex: 0 1 50%;
+}
+
+.cld-flex-col-3 {
+    flex: 0 1 33.3%;
+}
+
+.cld-flex-col-4 {
```

Comment:
Remove 25% in case there are more than 4 items.

Reply:
i will calc dynamically since the total elements in view have no limit